### PR TITLE
Improve clarity and readability of password input documentation

### PR DESCRIPTION
**Pull request by erbsland-dev**
- Refined descriptions for password input methods: `file:`, `fd:`, and `stdin`
- Enhanced readability and consistency in the instructions
- Clarified handling of newline and carriage return characters in password input

- [x] documentation is added or updated

### CODE REVIEW

**Review by michael-o — APPROVED** [review]
I wonder how this works on Windows then given that the `\r` is always present...

**Comment by erbsland-dev** [comment]
> I wonder how this works on Windows then given that the `\r` is always present...

I assume it does not work very well, but I haven’t found any issues or change requests related to this behavior so far.

**Comment by kroeckx** [comment]
If a file is open in text mode, a \r\n is converted to a \n on Windows, while in binary mode it's not.

**Comment by michael-o** [comment]
> If a file is open in text mode, a \r\n is converted to a \n on Windows, while in binary mode it's not.

FDs and stdin as well?

**Comment by erbsland-dev** [comment]
@kroeckx Good point. The library opens the file in text mode, so the `file:` option is not affected. This issue only arises with existing streams that are opened in binary mode.

**Comment by erbsland-dev** [comment]
@kroeckx I’ve rephrased the documentation slightly. The note about the carriage return has been moved to the `fd:` section, as it only applies there. The `file:` and `stdin` options are unaffected by this behavior because the library correctly opens these streams in text mode.

**Review by nhorman — APPROVED** [review]

**Comment by erbsland-dev** [comment]
@nhorman I accidentally typed `fixed` instead of `fixes` in the commit, it seems that didn't properly linked this PR to issue #7310.

**Comment by nhorman** [comment]
@erbsland-dev ack, fixed it up.  Thanks for the heads up!

File: doc/man1/openssl-passphrase-options.pod [hunk 1]
```diff
@@ -46,26 +46,33 @@ the environment of other processes is visible on certain platforms
...
 =item B<fd:>I<number>
 
-Read the password from the file descriptor I<number>. This can be used to
-send the data via a pipe for example.
+Reads the password from the file descriptor I<number>. This can be useful for
+sending data via a pipe, for example. The same line handling as described for
+B<file:> applies to passwords read from file descriptors.
+
+Please note that if the file was opened in binary mode, a trailing carriage
+return character is preserved and will be interpreted as part of the password.
```
Comment (t8m) [inline doc/man1/openssl-passphrase-options.pod:64]: Actually `fd:` is not supported on Windows and the CR character is not present on posixy platforms on EOL so please remove this and instead mention that `fd:` is not supported on Windows.

**Review by t8m — CHANGES_REQUESTED** [review]

File: doc/man1/openssl-passphrase-options.pod (hunk 1 above)
Comment (erbsland-dev) [reply doc/man1/openssl-passphrase-options.pod]: I changed this and rebased the PR to the latest master.

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
@nhorman please reconfirm

**Comment by nhorman** [comment]
Ack, approval holds

**Review by levitte — APPROVED** [review]

**Comment by t8m** [comment]
Merged to all the active branches. Thank you for your contribution.

**Pull request closed without merging**